src/csvcubed/cli/inspect/metadataprinter.py

```diff
@@ -150,24 +151,25 @@ def get_datacube_results(self):
 
         Member of :class:`./MetadataPrinter`.
         """
-        assert isinstance(self.state, DataCubeState)  # Make pyright happier
-
-        csvw_state = self.state.csvw_state
+        assert isinstance(self.state, DataCubeInspector)  # Make pyright happier
 
         self.result_qube_components = self.state.get_dsd_qube_components_for_csv(
             self.primary_csv_url
         )
 
-        self.result_dataset_label_dsd_uri = (
-            select_csvw_dsd_dataset_label_and_dsd_def_uri(csvw_state.rdf_graph)
+        self.result_cube_table_identifiers = self.state.get_cube_identifiers_for_csv(
+            self.primary_csv_url
         )
-        self.result_cols_with_suppress_output_true = (
-            select_cols_where_suppress_output_is_true(csvw_state.rdf_graph)
+        self.result_column_definitions = self.state.get_column_definitions_for_csv(
+            self.primary_csv_url
         )
-        self.result_code_lists = select_dsd_code_list_and_cols(
-            csvw_state.rdf_graph,
-            self.result_dataset_label_dsd_uri.dsd_uri,
-            csvw_state.csvw_json_path,
+
+        self.suppressed_columns = self.state.get_suppressed_columns_for_csv(
```

Comment:
I do wonder whether we need to store `suppressed_columns` separately since the information is already contained within `result_column_definintions`.

Reply:
I have left this as it is for now because I wanted to ask; does this mean we should get rid of the get_suppressed_columns_for_csv function? I agree we can definitely get what we need from the column definitions, just not sure if it is necessary or not.

Reply:
I think we should get rid of that function since we can derive the same information from the column definitions.